### PR TITLE
Create Development plan

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
Signed-off-by: TomSweeneyRedHat <tsweeney@redhat.com>

First pass for the Buildah development plan.  Also removed links from the Demos page that best live on the main page.

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
Forgot to add, addresses #718

File: developmentplan.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+![buildah logo](https://cdn.rawgit.com/projectatomic/buildah/master/logos/buildah-logo_large.png)
+
+# Development Plan
+
+## Development goals for Buildah
+
+ *  Integration into OpenShift and potentially other tools.  The biggest requirement for this is to be able build OCI Images without running as root on a system, actually no SYS_ADMIN privs.  This would allow Buildah to run non-privileged containers inside of Kubernetes, so you could distribute your container workloads.
+
+ * Integration with User Namespace, Podman has this already and the goal is to get `buildah bud` and `buildah run` to be able to run its containers in a usernamespace to give the builder bettter security isolation from the host.
```
Comment (rhatdan) [inline developmentplan.md]: bettter?

File: developmentplan.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+ * Caching support, currently `buildah bud` builds just one layer, which is useful for building images in a pipeline, but is not as useful for developers. The `docker build` command defaults to caching, where each line of the Dockerfile creates an image. If you don't modify a line in a Dockerfile and re-execute docker build a second time, it will skip all of the lines in the Dockerfile until a line changes. Similar support is planned for Buildah.
```
Comment (rhatdan) [inline developmentplan.md]: We already have this --layers.

File: developmentplan.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+![buildah logo](https://cdn.rawgit.com/projectatomic/buildah/master/logos/buildah-logo_large.png)
+
+# Development Plan
+
+## Development goals for Buildah
+
+ *  Integration into OpenShift and potentially other tools.  The biggest requirement for this is to be able build OCI Images without running as root on a system, actually no SYS_ADMIN privs.  This would allow Buildah to run non-privileged containers inside of Kubernetes, so you could distribute your container workloads.
```
Comment (rhatdan) [inline developmentplan.md]: The goal for OpenShift is not No-Rooot, it is no SYS_ADMIN, which means limited root, or running buildah in a standard linux container.  IE Not modifying any security constraints.   This requirement should mention Kubernetes not OpenShift.

Comment (TomSweeneyRedHat) [reply developmentplan.md]: I hacked this up a bit.  Please correct if it needs more hacking.

File: developmentplan.md (hunk 1 above)
Comment (TomSweeneyRedHat) [reply developmentplan.md]: bettter now be better.

File: developmentplan.md (hunk 2 above)
Comment (TomSweeneyRedHat) [reply developmentplan.md]: Removed this

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@umohnani8 @nalind @pixdrift PTAL

**Comment by TomSweeneyRedHat** [comment]
Weird ubuntu error with Golang 1.9.  Going to kick the bot again.

**Comment by TomSweeneyRedHat** [comment]
bot, retest this please.

**Comment by umohnani8** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
@pixdrift pointed out in IRC that we still listed 'alpha' in the README.md.  I've remove that line from the README.md and it's the only change with the latest push.

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**